src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -158,8 +159,8 @@ object JoinIndexRule
    */
   private def isPlanModified(plan: LogicalPlan): Boolean = {
     plan.find {
-      case LogicalRelation(fsRelation: HadoopFsRelation, _, _, _) =>
-        RuleUtils.isIndexApplied(fsRelation)
+      case p: LogicalRelation =>
```

Comment:
No need to support `DataSourceV2Relation`?

Reply:
👍 Good catch. Missed this one.

Reply:
As DataSourceV2Relation is not used in a transformed plan, we don't need to support it here.

Reply:
Changed the code to use the `LogicalPlanUtils.hasSupportedLogicalRelation` method.

Reply:
Reverting as per your suggestion.